File: common/src/main/java/generations/gg/generations/core/generationscore/world/item/legends/CreationTrioItem.java
package generations.gg.generations.core.generationscore.world.item.legends;

import generations.gg.generations.core.generationscore.client.model.ModelContextProviders;
import generations.gg.generations.core.generationscore.config.SpeciesKey;
import generations.gg.generations.core.generationscore.world.item.ItemWithLangTooltipImpl;
import net.minecraft.resources.ResourceLocation;

public class CreationTrioItem extends ItemWithLangTooltipImpl implements ModelContextProviders.ModelProvider {
    private final SpeciesKey speciesId;
    private final ResourceLocation model;

    public CreationTrioItem(Properties properties, SpeciesKey speciesId, ResourceLocation model) {
        super(properties);
        this.speciesId = speciesId;
        this.model = model;
    }

    @Override
    public ResourceLocation getModel() {
        return model;
    }

    public SpeciesKey getSpeciesId() {
        return speciesId;
    }
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/item/legends/CreationTrioItem.kt
package generations.gg.generations.core.generationscore.common.world.item.legends

import generations.gg.generations.core.generationscore.common.client.model.ModelContextProviders
import generations.gg.generations.core.generationscore.common.config.SpeciesKey
import generations.gg.generations.core.generationscore.common.world.item.ItemWithLangTooltipImpl
import net.minecraft.resources.ResourceLocation

class CreationTrioItem(properties: Properties, @JvmField val speciesId: SpeciesKey, private val model: ResourceLocation) :
    ItemWithLangTooltipImpl(properties), ModelContextProviders.ModelProvider {
    override fun getModel(): ResourceLocation {
        return model
    }
}
